from pyrogram import Client, filters
from pyrogram.errors import UserAlreadyParticipant
import asyncio
from helpers.decorators import authorized_users_only, errors
from callsmusic.callsmusic import client as USER
from config import SUDO_USERS


@Client.on_message(filters.command(["userbotjoin"]) & ~filters.private & ~filters.bot)
@authorized_users_only
@errors
async def addchannel(client, message):
    chid = message.chat.id
    try:
        invitelink = await client.export_chat_invite_link(chid)
    except:
        await message.reply_text(
            "<b>make me as admin first !</b>",
        )
        return

    try:
        user = await USER.get_me()
    except:
        user.first_name = "helper"

    try:
        await USER.join_chat(invitelink)
        await USER.send_message(message.chat.id, "ðŸ¤–: Saya telah bergabung untuk memutar musik di VCG anda!")
    except UserAlreadyParticipant:
        await message.reply_text(
            "<b>âœ… **assistant telah ada di grup**</b>",
        )
    except Exception as e:
        print(e)
        await message.reply_text(
            f"<b>ðŸ›‘ Flood Wait Error ðŸ›‘ \n {user.first_name} tidak bisa bergabung karena terlalu banyak permintaan bergabung kepada userbot! Pastikan assistant tidak terbanned di grup."
            "\n\nAtau tambahkan assistant secara manual dan coba lagi!</b>",
        )
        return
    await message.reply_text(
        "<b>**âœ… Assistant bergabung ke grup**\n\n**â€¢ Jika ada masalah bisa laporkan ke** @kenbotsupport</b>",
    )


@USER.on_message(filters.group & filters.command(["userbotleave"]))
@authorized_users_only
async def rem(USER, message):
    try:
        await USER.leave_chat(message.chat.id)
    except:
        await message.reply_text(
            f"<b>assistant tidak bisa keluar dari grup karena floodwait."
            "\n\nAnda bisa menggunakan kick manual dan coba lagi</b>",
        )
        return
    
@Client.on_message(filters.command(["userbotleaveall"]))
async def bye(client, message):
    if message.from_user.id in SUDO_USERS:
        left=0
        failed=0
        lol = await message.reply("Assistant Leaving all chats")
        async for dialog in USER.iter_dialogs():
            try:
                await USER.leave_chat(dialog.chat.id)
                left = left+1
                await lol.edit(f"Assistant leaving... Left: {left} chats. Failed: {failed} chats.")
            except:
                failed=failed+1
                await lol.edit(f"Assistant leaving... Left: {left} chats. Failed: {failed} chats.")
            await asyncio.sleep(0.7)
        await client.send_message(message.chat.id, f"Left {left} chats. Failed {failed} chats.")
